docs/guides/setup.md

```diff
@@ -1,124 +1,186 @@
-```js
-videojs("example_video_1", {}, function(){
-  // Player (this) is initialized and ready.
-});
+> **Note:** The example with `controls="false"` can be a point of confusion for new developers - it will actually turn controls _on_!
+
+These are correct:
+
+```html
+<video controls ...>
+<video loop="loop" ...>
+<video ...>
 ```
 
-The first argument in the `videojs` function is the ID of your video tag. Replace it with your own.
+## Player Readiness
 
-The second argument is an options object. It allows you to set additional options like you can with the data-setup attribute.
+Because Video.js techs have the potential to be loaded asynchronously, it isn't always safe to interact with a player immediately upon setup. For this reason, Video.js players have a concept of "readiness" which will be familiar to anyone who has used jQuery before.
 
-The third argument is a 'ready' callback. Once Video.js has initialized it will call this function.
+Essentially, any number of ready callbacks can be defined for a Video.js player. There are three ways to pass these callbacks. In each example, we'll add an identical class to the player:
 
-Instead of using an element ID, you can also pass a reference to the element itself.
+Pass a callback to the `videojs()` function as a third argument:
 
 ```js
-videojs(document.getElementById('example_video_1'), {}, function() {
-  // This is functionally the same as the previous example.
+// Passing `null` for the options argument.
+videojs('my-player', null, function() {
+  this.addClass('my-example');
 });
 ```
 
+Pass a callback to a player's `ready()` method:
+
 ```js
-videojs(document.getElementsByClassName('awesome_video_class')[0], {}, function() {
-  // You can grab an element by class if you'd like, just make sure
-  // if it's an array that you pick one (here we chose the first).
+var player = videojs('my-player');
+
+player.ready(function() {
+  this.addClass('my-example');
 });
 ```
 
-\* If you have trouble playing back content you know is in the [correct format](http://blog.zencoder.com/2013/09/13/what-formats-do-i-need-for-html5-video/), your HTTP server might not be delivering the content with the correct [MIME type](http://en.wikipedia.org/wiki/Internet_media_type#Type_video). Please double check your content's headers before opening an [issue](/CONTRIBUTING.md).
+Listen for the player's `"ready"` event:
+
+```js
+var player = videojs('my-player');
+
+player.on('ready', function() {
+  this.addClass('my-example');
+});
+```
+
+In each case, the callback is called asynchronously - _even if the player is already ready!_
```

Comment:
We don't have any way to make it synchronous. It was removed in 5.0, since previously, it depended on what the tech did, so, html5 was sync but flash was async. Now it's always async. However, #3782 makes sense. I think we may not want all 3, but maybe two of them.